### ISSUE TITLE
Can't build project with serde with latest cargo

### ISSUE THREAD

**Issue by ehuss** [rust-lang/cargo#5453]
Building a project with a dependency `serde = "1.0"` causes the following error:

```
error: no matching version `^1.0` found for package `serde`
location searched: registry `https://github.com/rust-lang/crates.io-index`
versions found: 0.9.6, 0.9.5, 0.9.4, ...
```

This might be caused by #5300 (cc @djc).

**Comment by alexcrichton** [comment]
Thanks for the report @ehuss! I've found the issue and am working on a test.

**Comment by alexcrichton** [comment]
This should be fixed at #5454
